src/index.ts

```diff
@@ -737,8 +766,33 @@ export function processProject(
     .getSourceFiles('./**/*.guard.ts')
     .forEach(sourceFile => sourceFile.delete())
 
+  const sourceFiles = project.getSourceFiles()
+  // Sort source files by dependencies - dependencies before dependants
+  const orderedSourceFiles: SourceFile[] = []
+  const orderSourceFileByDependencies = (sourceFile: SourceFile) => {
+    // Ignore if already added as a dependency of another
+    if (orderedSourceFiles.includes(sourceFile)) {
+      return
+    }
+    // Add all dependencies to the ordered list first (if they have beeen specified and have not already been added)
+    sourceFile.getImportDeclarations().forEach(importDeclaration => {
+      const importSourceFile = importDeclaration.getModuleSpecifierSourceFile()
+      if (
+        importSourceFile &&
+        sourceFiles.includes(importSourceFile) &&
+        !orderedSourceFiles.includes(importSourceFile)
+      ) {
+        orderSourceFileByDependencies(importSourceFile)
+      }
+    })
+    // Add this one to the ordered list
+    orderedSourceFiles.push(sourceFile)
+  }
+  sourceFiles.forEach(orderSourceFileByDependencies)
```

Comment:
This process may have issues if cyclical imports are used... if they are even possible? The fix would be to compile a list of "parents" that gets passed to each subsequent child recursion, if we see a file that's already in the list of parents we exit.

... let me just see if cyclical imports are actually possible or not

Reply:
Look like that is a possibility, I will add the fix to make sure it never causes an infinite loop